File: docs/snippets/all/archetypes/points2d_simple.cpp
// Log some very simple points.

#include <dalaran.hpp>

int main(int argc, char* argv[]) {
    const auto rec = dalaran::RecordingStream("dalaran_example_points2d");
    rec.spawn().exit_on_failure();

    rec.log("points", dalaran::Points2D({{0.0f, 0.0f}, {1.0f, 1.0f}}));

    // TODO(#5520): log VisualBounds2D
}

File: docs/snippets/all/archetypes/points2d_simple.rs
//! Log some very simple points.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec =
        dalaran::RecordingStreamBuilder::new("dalaran_example_points2d").spawn()?;

    rec.log("points", &dalaran::Points2D::new([(0.0, 0.0), (1.0, 1.0)]))?;

    // TODO(#5521): log VisualBounds2D

    Ok(())
}
